File: 1.TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // for(int i = 0; i < nums.length - 1; i++) {
        //     for(int j = i + 1; j < nums.length; j++) {
        //         if(target == nums[i] + nums[j]) {
        //             return new int[]{i, j};
        //         }
        //     }
        // }
        // return null;   
        
        Map<int, int> m = new HashMap();
        for(int i = 0; i < nums.length; i++) {
            int cur = nums[i];
            if(m.containKey(target - cur)) {
                return new int[]{m.get(target - cur), i};
            } else {
                m.put(cur, i);
            }
        }
        return null;
    }
}

File: 1.TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        /*
        vector<int> ret;
        for(int i = 0; i < nums.size(); i++) {
            for(int j = i + 1; j < nums.size(); j++) {
                if(target == nums[i] + nums[j]) {
                    return ret = {i, j};
                }
            }
        }
        return ret;
        */
        
        
        map<int, int> m;
        map<int, int>::iterator iter;
        for(int i = 0; i < nums.size(); i++) {
            iter = m.find(target - nums[i]);
            if(iter == m.end()) {
                //추가
                m[nums[i]] = i;
            } else {
                //찾음
                return {(*iter).second, i};
            }
        }
        return {};
    }
};